lib/utils/datasetManager.cpp

```diff
@@ -25,7 +25,7 @@ dataset::dataset(json& js) {
         _base_dset = 0;
     else
         _base_dset = js["base_dset"];
-    _types = js["types"].get<std::set<std::string>>();
+    _type = js["type"].get<std::string>();
```

Comment:
Do we still need it such that the broker knows which states to ask for if it doesn't have them locally?

Reply:
I'm not sure if I unerstand what you mean.
If a state ID gets registered and comet doesn't know it, it will ask for the state.

comet doesn't use the `types` field in datasets for anything.